core/banking/recurring_transfers.go

```diff
@@ -49,6 +49,16 @@ func (e *Engine) recurringTransfer(
 		return fmt.Errorf("could not transfer funds: %w", err)
 	}
 
+	if transfer.DispatchStrategy != nil {
+		// ensure the asset transfer is correct
+		_, err := e.assets.Get(transfer.DispatchStrategy.AssetForMetric)
```

Comment:
I don't think this is correct. This may be legit empty if the dispatch strategy is market proposer

Reply:
right. let me fix.